src/main/java/com/aws/greengrass/authorization/AuthorizationHandler.java

```diff
@@ -419,12 +421,19 @@ private void addPermission(String destination,
                             Permission.builder().principal(principal).operation(operation).resource(null).build());
                 } else {
                     for (String resource : resources) {
-                        authModule.addPermission(destination,
-                                Permission.builder()
-                                        .principal(principal)
-                                        .operation(operation)
-                                        .resource(resource)
-                                        .build());
+                        try {
+                            authModule.addPermission(destination,
+                                    Permission.builder()
+                                            .principal(principal)
+                                            .operation(operation)
+                                            .resource(resource)
+                                            .build());
+                        } catch (AuthorizationException e) {
+                            logger.atError("load-authorization-config-add-resource-error").setCause(e)
+                                    .log("PolicyId {}: Error while adding permission for principal {} "
```

Comment:
we want to include the destination too. Make it impossible for a customer to not know where to look to fix it

Reply:
And don't necessarily use the terms which we use in code, this needs to be customer friendly.

Reply:
policyId should tell them where to look. Anyways I can add it, no harm in it.
Also, Can I use the term service for the Destination Component?
`PolicyId {}: Error while adding permission for component {principal} to perform operation {op} ({destination}) on resource {resource}`

Reply:
https://docs.aws.amazon.com/greengrass/v2/developerguide/interprocess-communication.html#ipc-authorization-policies
We use "IPC service" in the docs it seems